core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -1746,20 +1764,20 @@ public function edit($intId=null, $ajaxId=null)
 			$this->intId = $intId;
 		}
 
-		$this->denyAccessUnlessGranted(ContaoCorePermissions::DC_ACTION_EDIT, new DataContainerSubject($this->strTable, $this->intId));
-
 		// Get the current record
-		$objRow = $this->Database->prepare("SELECT * FROM " . $this->strTable . " WHERE id=?")
-								 ->limit(1)
-								 ->execute($this->intId);
+		$this->loadActiveRecord();
 
 		// Redirect if there is no record with the given ID
-		if ($objRow->numRows < 1)
+		if (null === $this->activeRecord)
 		{
 			throw new AccessDeniedException('Cannot load record "' . $this->strTable . '.id=' . $this->intId . '".');
 		}
 
-		$this->objActiveRecord = $objRow;
+		// TODO: This should get $new but we can only do that once we have solved the save() issue
```

Comment:
there should be a secondary permission check with the new data, but pre-checking the original record here would be correct to me

Reply:
I guess this comment is obsolete now.